app/client/jsx/Welcome.jsx

```diff
@@ -8,127 +8,109 @@ import { HttpApi } from './WebAPI.jsx'
+      if (room.art && room.art.src) {
+        const img = new Image()
+        img.src = room.art.src
+      }
+    })
+  }
 
-    handleUsernameChange(event) {
-        this.setState({ username: event.target.value })
-    }
+  componentDidMount () {
+    this.fetchRooms()
+  }
 
-    handleAvatarSelect(selection) {
-        this.setState({ avatar: selection })
-    }
+  handleUsernameChange (event) {
+    this.setState({ username: event.target.value })
+  }
 
-    async handleReady() {
-        const response = await this.httpApi.join(this.state.username, this.state.avatar)
-        if (response.success) {
-            const { username, avatar } = this.state
-            const user = {
-                username,
-                avatar,
-                userId: response.userId
-            }
-            this.props.updateUser(user)
-            this.handleUserSelected(user)
-            this.setState({redirect: '/party'})
-        }
-    }
+  handleAvatarSelect (selection) {
+    this.setState({ avatar: selection })
+  }
 
-    handleUserSelected(user) {
-        LocalStorage.set("USER", user)
-        this.props.updateUser(user)
-        this.props.updateCurrentRoom({
-            room: 'vestibule',
-            entered: false
-        })
+  async handleReady () {
+    const response = await this.httpApi.join(this.state.username, this.state.avatar)
+    if (response.success) {
+      const { username, avatar } = this.state
+      const user = {
+        username,
+        avatar,
+        userId: response.userId
+      }
+      this.props.updateUser(user)
+      this.handleUserSelected(user)
+      this.setState({ redirect: '/party' })
     }
+  }
 
-    render() {
-        if (this.state.redirect) {
-            return <Redirect to={this.state.redirect}/>
-        }
-
-        let text_entry = {
-          padding: '6px 10px',
-          border: '2px solid #D6D3CD',
-          backgroundColor: '#D6D3CD',
-          borderRadius: '7px',
-          color: '#1B1E1F',
-          outline: 'none',
-          boxShadow: 'none',
-          fontSize: '20px'
-        }
+  handleUserSelected (user) {
+    LocalStorage.set ('USER', user)
+    this.props.updateUser(user)
+    this.props.updateCurrentRoom({
+      room: 'vestibule',
+      entered: false
+    })
+  }
 
-        let name_opacity = 'form-fade'
-        let avatar_opacity = 'form-fade'
-        let party_opacity = 'form-fade-party'
-        if (this.state.username === null) { name_opacity = 'form' }
-        if (this.state.username) { avatar_opacity = 'form' }
-        if (this.state.username && this.state.avatar) { party_opacity = 'form-party' }
+  render () {
+    if (this.state.redirect) { return <Redirect to={this.state.redirect} /> }
+    const nameOpacity = (this.state.username === null) ? 'form' : 'form-fade'
+    const avatarOpacity = (this.state.username) ? 'form' : 'form-fade'
+    const partyOpacity = (this.state.username && this.state.avatar) ? 'form-party' : 'form-fade-party'
```

Comment:
Is there a way to combine these? It looks like you evaluate `this.state.username` several times; maybe this can be done once